src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -550,7 +595,29 @@ static BoundStatement generateCopyTo(SyntheticBoundNodeFactory f, MethodSymbol m
                 var parameterReference0 = f.Parameter(method.Parameters[0]);
                 var parameterReference1 = f.Parameter(method.Parameters[1]);
                 BoundStatement statement;
-                if (!interfaceMethod.ContainingType.IsGenericType || field.Type.IsArray())
+                if (containingType.IsSingleton)
+                {
+                    if (!interfaceMethod.ContainingType.IsGenericType)
+                    {
+                        var arraySetValueMethod = (MethodSymbol)method.DeclaringCompilation.GetWellKnownTypeMember(WellKnownMember.System_Array__SetValue)!;
```

Comment:
>WellKnownMember.System_Array__SetValue [](http://example.com/codeflow?start=115&length=38)

We should verify this member is available in the `Create` method, similar to the specific members for `SynthesizedReadOnlyListKind.List`  in `s_requiredWellKnownMembersUnknownLength`. And consider adding a corresponding test, similar to `SynthesizedReadOnlyList_MissingWellKnownMembers_UnknownLength`.

Reply:
Correct me if I'm wrong, but since this is a nested type there's no Create method -- we don't want to return an error type if any type/member is missing in which case we fallback to Array kind instead which is the current behavior.

I'll try that in the next iteration.